src/rtcModule/webrtc.cpp

```diff
@@ -716,16 +816,35 @@ void Call::msgCallReqCancel(RtMessage& packet)
 
 void Call::handleReject(RtMessage& packet)
 {
-    if (mState != Call::kStateReqSent && mState != Call::kStateInProgress)
+    if (packet.callid != mId)
     {
-        SUB_LOG_WARNING("Ingoring unexpected CALL_REJECT while in state %s", stateToStr(mState));
+        SUB_LOG_WARNING("Ingoring unexpected call id");
         return;
     }
-    if (mIsGroup || !mSessions.empty())
+
+    if (packet.userid != mChat.client().userId())
     {
-        return;
+        if (mState != Call::kStateReqSent && mState != Call::kStateInProgress)
+        {
+            SUB_LOG_WARNING("Ingoring unexpected CALL_REQ_DECLINE while in state %s", stateToStr(mState));
+            return;
+        }
+        if (mIsGroup || !mSessions.empty())
+        {
+            return;
+        }
+        destroy(static_cast<TermCode>(TermCode::kCallRejected | TermCode::kPeer), false);
+    }
+    else // Call has been rejected by other client from same user
+    {
```

Comment:
Here maybe it's appropriate to have an assertion to check the clientid from received packet is different than ours.